### DISCUSSION TITLE
Native module cannot be null

### DISCUSSION THREAD

**Discussion by ErikTromp** [hyochan/react-native-iap#1639] [Q&A]
I did a fresh install of react-native-iap (version 8.0.1) in my Expo RN project and am getting this error. Any idea how to fix this?

```
Invariant Violation: Native module cannot be null.
at node_modules\react-native\Libraries\LogBox\LogBox.js:148:8 in registerError
at node_modules\react-native\Libraries\LogBox\LogBox.js:59:8 in errorImpl
at node_modules\react-native\Libraries\LogBox\LogBox.js:33:4 in console.error
at node_modules\expo\build\environment\react-native-logs.fx.js:27:4 in error
at node_modules\react-native\Libraries\Core\ExceptionsManager.js:104:6 in reportException
at node_modules\react-native\Libraries\Core\ExceptionsManager.js:171:19 in handleException
at node_modules\react-native\Libraries\Core\setUpErrorHandling.js:24:6 in handleError
at node_modules\expo-error-recovery\build\ErrorRecovery.fx.js:12:21 in ErrorUtils.setGlobalHandler$argument_0
at node_modules\regenerator-runtime\runtime.js:63:36 in tryCatch
at node_modules\regenerator-runtime\runtime.js:294:29 in invoke
at node_modules\regenerator-runtime\runtime.js:63:36 in tryCatch
at node_modules\regenerator-runtime\runtime.js:155:27 in invoke
at node_modules\regenerator-runtime\runtime.js:165:18 in PromiseImpl.resolve.then$argument_0
at node_modules\react-native\node_modules\promise\setimmediate\core.js:37:13 in tryCallOne
at node_modules\react-native\node_modules\promise\setimmediate\core.js:123:24 in setImmediate$argument_0
at node_modules\react-native\Libraries\Core\Timers\JSTimers.js:130:14 in _callTimer
at node_modules\react-native\Libraries\Core\Timers\JSTimers.js:181:14 in _callImmediatesPass
at node_modules\react-native\Libraries\Core\Timers\JSTimers.js:441:30 in callImmediates
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:387:6 in __callImmediates
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:135:6 in __guard$argument_0
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:364:10 in __guard
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:134:4 in flushedQueue
```

**Comment by idlework** [comment]
@ErikTromp This error log does not show anything related to `react-native-iap`, do you have an other error message? Also upgrading `react-native-iap` to the latest release `8.0.5` maybe it fixes your issue :D

**Comment by ErikTromp** [comment]
Yeah correct, just for others looking here: I was using Expo with dev-client and didn't have a plugin configured properly. Defining the plugin and recompiling client worked

**Reply by mfkrause** [reply]
@ErikTromp Could you share the config plugin you've written? Maybe this could be integrated directly into react-native-iap, too.

**Reply by mekwall** [reply]
@ErikTromp I would like to know as well what you did to get it to work 😸

**Reply by ErikTromp** [reply]
I am sorry but have to disappoint you. I remember I got the plugin and this module to work but we ran into other issues with the library... In the current version of our app we have ejected Expo and switched to `expo-in-app-purchases` which, despite being Expo, doesn't work with Expo managed flow but does work fine when Expo is ejected. We have not encountered a single problem with that library, so we stuck with it (it also doesn't require to add any additional config plugins)

**Reply by mekwall** [reply]
Thanks @ErikTromp! It helps since we are having similar issues so will probably follow suit 😅
